lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -29,7 +29,19 @@
 
 <li class="course-item">
   % if settings.FEATURES.get('ENABLE_VERIFIED_CERTIFICATES'):
-    <article class="course ${enrollment.mode}">
+    % if enrollment.mode == "verified":
+      % if settings.FEATURES.get('SEPARATE_VERIFICATION_FROM_PAYMENT') and request.session.get('separate-verified', False):
+        % if verification_status.get('status') in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_APPROVED]:
+          <article class="course verified">
+        % else:
+          <article class="course honor">
+        % endif
+      % else:
+        <article class="course verified">
+      % endif
+    % else:
+      <article class="course ${enrollment.mode}">
+    % endif
```

Comment:
This makes sense to me, but I think you'll need to handle the "professional" case as well.

Could you please add test cases to `student/tests/test_verification_status`?  I think this logic is sufficiently complex (and easy enough to break) that we should be testing it.